tests/authentication_tests/test_interactive_authentication.py

```diff
@@ -86,38 +86,41 @@ def _prepare_window(self, ld):
         QtGui.QApplication.processEvents()
 
     @contextlib.contextmanager
-    def _login_dialog(self, is_session_renewal, login=None, hostname=None):
+    def _login_dialog(self, is_session_renewal, hostname=None):
         # Import locally since login_dialog has a dependency on Qt and it might be missing
         from tank.authentication import login_dialog
 
-        with contextlib.closing(
-            login_dialog.LoginDialog(is_session_renewal=is_session_renewal)
-        ) as ld:
-            # SSO module and threading causes issues with unit tests, so disable it.
-            ld._saml2_sso = None
-            self._prepare_window(ld)
-            yield ld
+        # Patch out the SsoSaml2Toolkit class to avoid threads being created, which cause
+        # issues with tests.
+        with patch("tank.authentication.login_dialog.SsoSaml2Toolkit"):
+            with contextlib.closing(
+                login_dialog.LoginDialog(
+                    is_session_renewal=is_session_renewal, hostname=hostname
```

Comment:
Note that the test didn't use to pass in hostname, which means the test was not testing quite a few things. Now it does.